### PR TITLE
Fix PHP8 deprecations

### PR DESCRIPTION
**Pull request by jaykay-design**
Added the return types for the Iterator overrides

### CODE REVIEW

File: src/AssetCollection.php [hunk 1]
```diff
@@ -139,32 +139,32 @@ public function remove($name)
...
     }
 
-    public function valid()
+    public function valid(): bool
     {
         return isset($this->items[$this->index]);
     }
 
-    public function current()
+    public function current(): mixed
```
Comment (markstory) [inline src/AssetCollection.php]: I don't think we can use `mixed` yet as it causes errors in PHP7.x. We might be able to use `#[\ReturnTypeWillChange]` attribute and no formal return type.

Comment (jaykay-design) [reply src/AssetCollection.php]: Thanks for that. I've added the attributes and remove the mixed return type.

**Comment by dereuromark** [comment]
I have similar errors now on my end:
```
PHP Deprecated:  Return type of MiniAsset\AssetCollection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/markstory/mini-asset/src/AssetCollection.php on line 142
PHP Deprecated:  Return type of MiniAsset\AssetCollection::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/markstory/mini-asset/src/AssetCollection.php on line 167
PHP Deprecated:  Return type of MiniAsset\AssetCollection::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/markstory/mini-asset/src/AssetCollection.php on line 152
PHP Deprecated:  Return type of MiniAsset\AssetCollection::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/markstory/mini-asset/src/AssetCollection.php on line 157
PHP Deprecated:  Return type of MiniAsset\AssetCollection::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/markstory/mini-asset/src/AssetCollection.php on line 162
PHP Deprecated:  Return type of MiniAsset\AssetCollection::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/markstory/mini-asset/src/AssetCollection.php on line 147
```

**Comment by dereuromark** [comment]
Is there any plan on releasing this?
We face this now also in our default env (local dev is now 8.1)

**Pull request merged**